gum/gummemory.c

```diff
@@ -263,54 +288,106 @@ gum_memory_scan (const GumMemoryRange * range,
 }
 
 GumMatchPattern *
-gum_match_pattern_new_from_string (const gchar * match_str)
+gum_match_pattern_new_from_string (const gchar * match_combined_str)
 {
-  GumMatchPattern * pattern;
+  GumMatchPattern * pattern = NULL;
+  gchar ** splitted;
+  const gchar * match_str, * mask_str;
+  gboolean has_mask = FALSE;
   GumMatchToken * token = NULL;
-  const gchar * ch;
+  const gchar * ch, * mh;
+
+  splitted = g_strsplit (match_combined_str, ":", 2);
```

Comment:
Nit: `split` would be grammatically better, but is also a verb so maybe it's better to name this `parts`?